knowledge-base/grid-show-confirm-cancel-dialog.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Show Cancel Confirmation Dialog before proceeding with other data operations
+description: How to prompt the user to confirm canceling the edit before proceeding with other data operations?
+type: how-to
+page_title: Show Cancel Confirmation Dialog before proceeding with other data operations
+slug: grid-kb-show-confirm-cancel-dialog
+position: 
+tags: grid, edit, cancel, confirm, dialog
+ticketid: 1552190
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Grid for Blazor <br/> TreeList for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+Is there any way to show a cancel confirmation dialog when edit or insert operation is in progress and sorting/filtering/grouping option is initiated? For example, display a dialog with "Are you sure you want to cancel editing?" question and a confirmation button.
+
+How to prompt the user to confirm canceling the edit/create before proceeding with other data operations(sort, filter, group etc.)?
+
+## Solution
+
+If the user tries to perform some other operation (such as sorting or filtering) while the Grid is in create or edit mode, the creating/editing will be canceled (check the last bullet of the [Editing Notes section]({%slug components/grid/editing/overview%}#notes) for more details). The `OnCancel` event will fire and you may handle it to display a Dialog to allow the user choose whether they want to cancel the editing or not.
+
+We provide predefined and custom dialogs and based on which one you want to proceed with, the implementation for achieving the desired scenario will be different:
```

Comment:
The second sentence does not provide additional value at this point and can be removed.

```suggestion
Telerik UI for Blazor provides predefined and custom Dialogs:
```